src/nemo-main-application.c

```diff
@@ -391,21 +392,73 @@ open_tabs (NemoMainApplication *application,
     }
 }
 
+static void
+open_tabs_in_existing_window (NemoMainApplication *application,
+                              GFile **locations,
+                              guint n_files,
+                              GdkScreen *screen,
+                              const char *geometry)
+{
+    gchar *uri;
+
+    GList *list_copy;
+    GList *l;
+
+    list_copy = g_list_copy (gtk_application_get_windows (GTK_APPLICATION (&application->parent)));
+    for (l = list_copy; l != NULL; l = l->next) {
+        if (NEMO_IS_WINDOW (l->data)) {
+            NemoWindow *window;
+
+            window = NEMO_WINDOW (l->data);
+
+            /* open all locations */
+            for (int i = 1; i <= n_files; i++) {
+                /* open tabs in reverse order because each
+                 * tab is opened before the previous one */
+                guint tab = n_files - i;
+                uri = g_file_get_uri (locations[tab]);
+                g_debug ("Opening new tab at uri %s\n", uri);
+                nemo_window_go_to_tab (window, locations[tab]);
+                g_free(uri);
+            }
+
+            /* go to the last tab we opened */
+            NemoWindowPane *pane;
+
+            pane = nemo_window_get_active_pane (window);
+            nemo_notebook_set_current_page_relative (NEMO_NOTEBOOK (pane->notebook), n_files);
+
+            gtk_window_present (GTK_WINDOW (window));
```

Comment:
can you use gtk_window_present_with_time instead?

Maybe like this:
```
// you'll have to add #include <gdk/gdkx.h> to the top of this file

gtk_window_present_with_time (GTK_WINDOW (window),
                              gdk_x11_get_server_time (gtk_widget_get_window (GTK_WIDGET (window))));

```
The reason being, window managers decide whether or not to actually give a window focus by the time of this call compared with their last 'user' event (like a click or keypress). Using gtk_window_present really calls _present_with_time but with a time of 0. The window manager will ignore this window's focus request and try to just mark it urgent instead (flashing in the window list for example).

If you instead provide a real time, the wm thinks it was the user asking for this window and it's ok to raise this window to the top. GdkEvents (from clicks and such) usually provide this time for us, but in this case, though this is probably caused by the user, we need to get the time ourselves.

This has been broken forever with Cinnamon - any window has been able to raise, so your current form works there. I recently fixed this, however, so right now on my cinnamon/muffin master, when I run with `--existing-window` it doesn't actually present the window, it just marks it urgent.

See:
https://github.com/linuxmint/muffin/commit/ee9a615d75cc27734400fb3ae54b923ec6995ee6